### PR TITLE
Added Easy, Medium, and Hard Levels (fixes #6)

### LINKED ISSUE #6: Add Easy, Medium, and Hard Levels

### PR DESCRIPTION
**Pull request by Dendejar**
Created menu for selection of difficulty (Easy, Medium, Hard)
Easy - 2x2
Medium - 4x4
Hard - 6x6

### CODE REVIEW

**Comment by Dendejar** [comment]
other commits was mistakes, sorry but fixed it

File: src/main/java/board/GameMainView.java [hunk 1]
```diff
@@ -3,28 +3,81 @@
...
+    
     private static int rows = 4;
     private static int cols = 4;
     private static board.BoardView boardView = new board.BoardView(rows, cols);
-    private final JPanel panel = new JPanel(new BorderLayout());
-
-    public GameMainView(){
+    private static final JPanel panel = new JPanel(new BorderLayout());
+    
+    private static final JMenuBar mnubarMain = new JMenuBar();
```
Comment (scohen40) [inline src/main/java/board/GameMainView.java]: Please add in 'e's to spell Menu. Refactor/rename to be menuBarMain().

File: src/main/java/board/GameMainView.java [hunk 2]
```diff
@@ -3,28 +3,81 @@
...
+                    boardView.rows = 6;
+                    boardView.cols = 6;
+                    boardView.init();
+                    break;
+                default:
+                    break;
+            }
+                
+        }
+
```
Comment (scohen40) [inline src/main/java/board/GameMainView.java]: This code does not compile. There are errors in every board view statement.

File: src/main/java/board/GameMainView.java [hunk 3]
```diff
@@ -3,28 +3,81 @@
...
 public class GameMainView extends JFrame {
-
+    
     private static int rows = 4;
     private static int cols = 4;
     private static board.BoardView boardView = new board.BoardView(rows, cols);
-    private final JPanel panel = new JPanel(new BorderLayout());
-
-    public GameMainView(){
+    private static final JPanel panel = new JPanel(new BorderLayout());
```
Comment (scohen40) [inline src/main/java/board/GameMainView.java]: Take out all of the static identifiers.

File: src/main/java/board/GameMainView.java [hunk 4]
```diff
@@ -3,28 +3,81 @@
...
+    
+    private static final JMenuBar mnubarMain = new JMenuBar();
+    
+    private static final JMenu mnuDifficulty = new JMenu("Difficulty");
```
Comment (scohen40) [inline src/main/java/board/GameMainView.java]: Refactor to menuDifficulty()

File: src/main/java/board/GameMainView.java [hunk 5]
```diff
@@ -3,28 +3,81 @@
...
+    private static final String[] difficulty = {"Easy", "Medium", "Hard"};
+    private static final JMenuItem mnuitmDifficulty[] = new JMenuItem[difficulty.length];
```
Comment (scohen40) [inline src/main/java/board/GameMainView.java]: Refactor to menuItemDifficulty[]

File: src/main/java/board/GameMainView.java [hunk 6]
```diff
@@ -3,28 +3,81 @@
...
+        public DifficultyAction(int item) {
+            this.item = item;
+            
+        }
+
+        @Override
+        public void actionPerformed(ActionEvent e) {
+            switch (item) {
+                case 0:
+                    boardView.rows = 2;
```
Comment (scohen40) [inline src/main/java/board/GameMainView.java]: This does not compile, nor do any of the other boardview statements.

File: src/main/java/board/GameMainView.java [hunk 7]
```diff
@@ -3,28 +3,81 @@
...
+    
+    private static final JMenu mnuDifficulty = new JMenu("Difficulty");
+    private static final String[] difficulty = {"Easy", "Medium", "Hard"};
+    private static final JMenuItem mnuitmDifficulty[] = new JMenuItem[difficulty.length];
+    
+    public GameMainView(){       
         setTitle("MEMORY GAME");
         setSize(900, 750);
         setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
         setLocation(340, 80);
```
Comment (scohen40) [inline src/main/java/board/GameMainView.java]: Delete this line as it's overwritten in a few lines.

File: src/main/java/board/GameMainView.java [hunk 8]
```diff
@@ -3,28 +3,81 @@
...
-    public static void main(String[] args) {
+    public static void main(String[] args) {      
         new GameMainView().setVisible(true);
+       
+    }
+    
+    private static class DifficultyAction implements ActionListener{
+        int item = 0;
+        
+        public DifficultyAction(int item) {
```
Comment (scohen40) [inline src/main/java/board/GameMainView.java]: Instead of passing the item pass the size.

File: src/main/java/board/GameMainView.java [hunk 9]
```diff
@@ -3,28 +3,81 @@
...
+            this.item = item;
+            
+        }
+
+        @Override
+        public void actionPerformed(ActionEvent e) {
```
Comment (scohen40) [inline src/main/java/board/GameMainView.java:57]: get rid of the switch statement. you don't need it if you are passing in the size.

File: src/main/java/board/GameMainView.java [hunk 10]
```diff
@@ -3,30 +3,64 @@
...
+        int size;
+        
+        public DifficultyAction(int size){
+            this.size = size;
+            
+        }
+        
+        @Override
+        public void actionPerformed(ActionEvent e) {
+            boardView.rows = size;
```
Comment (scohen40) [inline src/main/java/board/GameMainView.java]: This does not compile. boardView.rows and boardView.cols are private.

**Comment by Dendejar** [comment]
Added resizeBoard() to BoardView.java

**Pull request merged**